client/src/main/java/io/getstream/chat/android/client/models/Channel.kt

```diff
@@ -52,7 +52,7 @@ public data class Channel(
     @IgnoreSerialisation
     @IgnoreDeserialisation
     var hiddenMessagesBefore: Date? = null,
-    val cooldown: Int = 0
+    var cooldown: Int = 0
```

Comment:
Why do we need this property as a mutable field?

Reply:
We're setting its value here, on the other PR: https://github.com/GetStream/stream-chat-android-livedata/pull/84/files#diff-03baeb758bf51b16e79734f427c8dd04cb657f7f8b231a082529c5e88e3cc991R59

Reply:
Yes, but it could be set by constructor, we don't need a mutable property for that

Reply:
True, but I think fixing the way we create instances of these models is out-of-scope for this PR, it's a much larger issue that we should tackle separately.

Reply:
What is the difference between:
```
val c = Channel()
c.cooldown = cooldown
// Rest of fields
```
and
```
val c = Channel(cooldown = cooldown)
// Rest of fields
```
I think it doesn't affect the scope of the other PR and we don't introduce an extra mutable property to our model that is not needed at all

Reply:
Following the pattern of the existing implementation - which unfortunately uses mutable properties - makes the code more consistent. It's easier to navigate the code if you see that all of these properties here are `var`s. You don't have to go _"Okay, so some of the properties are `val`s that I have to set in the ctor, but some of them are `var`s that I have to set after I've created a `Channel`..."_.

Adhering to the existing pattern simplifies the code. Having this pattern for the entire class is a problem we have to fix, but we don't really get closer to a fix by adding new properties in a different way. In the meantime, we just complicate the usage of the class.

Boring code that's all bad in the same, consistent, predictable way > partially fixed code which is more unpredictable.

Reply:
Just noticed two other `var`s here: I would actually change those too until we fix the root problem, so that this class has fewer surprises while working with it in the meantime 🙈

Reply:
I've done that for consistency with the rest of the code in livedata library. Closing this PR as the change is no longer needed.